src/main/drivers/bus_spi.c

```diff
@@ -17,89 +17,48 @@
 
 #include <stdbool.h>
 #include <stdint.h>
+#include <string.h>
 
 #include <platform.h>
 
+#include "build/debug.h"
+
 #include "drivers/bus_spi.h"
+#include "drivers/bus_spi_impl.h"
```

Comment:
The idea was to export only public interface from `bus_spi.h` and place declarations that are related to SPI implementation into `_imph.h` file (things that would be in .c file if SPI is implemented in single file).
For example SPI_device should be opaque from user code perspective. Using _impl.h is way to enforce this